docs/source/newsfragments/3578.change.2.rst

```diff
@@ -1 +1 @@
-The :envvar:`TESTCASE` variable has been changed to be a comma-separated list of regular expressions which match against test names to run. Each element of :envvar:`TESTCASE` can now match more than one test across all :envvar:`MODULE`\ s instead of just the first one found.
+The :envvar:`TESTCASE` variable has been changed so each element of :envvar:`TESTCASE` now matches all tests that match the test name across all :envvar:`MODULE`\ s instead of just the first one found.
```

Comment:
```suggestion
Use of the :envvar:`TESTCASE` variable has been changed so that each element of :envvar:`TESTCASE` will now select all tests with a matching name across all :envvar:`MODULE`\ s instead of just the first one found.
```